src/plugins/vis_type_xy/public/vis_component.tsx

```diff
@@ -307,10 +307,38 @@ const VisComponent = (props: VisComponentProps) => {
     ]
   );
   const xAccessor = getXAccessor(config.aspects.x);
-  const splitSeriesAccessors = config.aspects.series
-    ? compact(config.aspects.series.map(getComplexAccessor(COMPLEX_SPLIT_ACCESSOR)))
-    : [];
 
+  const splitSeriesAccessors = useMemo(
```

Comment:
This is irrelevant with the bug but it is an improvement to avoid unnecessary renders so I have included it here.